src/sqlfluff/core/rules/std.py

```diff
@@ -3462,11 +3462,14 @@ def _get_indexes(segment):
                 cnt_select_targets += 1
                 if first_select_target_idx == -1:
                     first_select_target_idx = fname_idx
-            if seg.is_type("keyword") and seg.name == "SELECT" and select_idx - 1:
+            if seg.is_type("keyword") and seg.name == "SELECT" and select_idx == -1:
```

Comment:
This is not directly related to the original issue, but the code just looks wrong. I'm pretty sure the intent was to check for a `-1` value, i.e. "not yet set" just as with `first_new_line_idx`.